Changelog fails when handling compare URLs if release is missing

**Command**: release
**Version**: 0.4.1

## What happens

When trying to call `release` command it fails if we have a link but not the corresponding version in file.

## Stack trace

```
panic: runtime error: index out of range [1] with length 0
goroutine 1 [running]:
github.com/rcmachado/changelog/chg.(*Changelog).Release(0xc000096c30, 0x7ffd5b806c9f, 0x5, 0xc0000206e0, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
    /go/src/github.com/rcmachado/changelog/chg/changelog.go:107 +0x689
github.com/rcmachado/changelog/cmd.newReleaseCmd.func1(0xc0000c5180, 0xc0000aa230, 0x1, 0x5, 0x0, 0x0)
    /go/src/github.com/rcmachado/changelog/cmd/release.go:40 +0x198
github.com/spf13/cobra.(*Command).execute(0xc0000c5180, 0xc0000aa1e0, 0x5, 0x5, 0xc0000c5180, 0xc0000aa1e0)
    /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0x7abf20, 0x43b60a, 0x797340, 0xc000000180)
    /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
    /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/rcmachado/changelog/cmd.Execute()
    /go/src/github.com/rcmachado/changelog/cmd/root.go:83 +0x31
main.main()
    /go/src/github.com/rcmachado/changelog/main.go:6 +0x20
```

## Changelog file

```markdown
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/)
and this project adheres to [Semantic Versioning](http://semver.org/).

## [Unreleased]
### Changed
- Do not exclude outliers when predicting.
- Adjusting CPU/memory for the endpoint.
- Configured Redis.

[Unreleased]: https://github.com/org/repo/compare/0.2.0...HEAD
[0.2.0]: https://github.com/org/repo/compare/0.1.0...0.2.0
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Fixed
+- Compare URL now parses versions with dots correctly
+
+## [0.4.1] - 2020-02-27
+### Fixed
+- Wrong version for blackfriday dependency
 
 ## [0.4.0] - 2020-02-27
 ### Fixed
@@ -38,7 +44,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - `fmt` command to reformat changelog following the spec
 - `show` command to show a specific version
 
-[Unreleased]: https://github.com/rcmachado/changelog/compare/0.4.0...HEAD
+[Unreleased]: https://github.com/rcmachado/changelog/compare/0.4.1...HEAD
+[0.4.1]: https://github.com/rcmachado/changelog/compare/0.4.0...0.4.1
 [0.4.0]: https://github.com/rcmachado/changelog/compare/0.3.0...0.4.0
 [0.3.0]: https://github.com/rcmachado/changelog/compare/0.2.0...0.3.0
 [0.2.0]: https://github.com/rcmachado/changelog/compare/0.1.0...0.2.0

--- a/chg/changelog.go
+++ b/chg/changelog.go
@@ -101,7 +101,7 @@ func (c *Changelog) Release(newVersion Version) (*Version, error) {
 		compareURL = strings.Replace(newVersion.Link, "<prev>", newVersion.Name, -1)
 		compareURL = strings.Replace(compareURL, "<next>", "HEAD", -1)
 	} else if prevVersion == nil || prevVersion.Link == "" {
-		r := regexp.MustCompile(`/(\w+)\.{2,3}(\w+)$`)
+		r := regexp.MustCompile(`(\w[\w\.]*?)\.{2,3}(\w[\w\.]*?)$`)
 		r.MatchString(oldUnreleased.Link)
 		matches := r.FindStringSubmatch(oldUnreleased.Link)
 		compareURL = strings.Replace(oldUnreleased.Link, matches[1], newVersion.Name, -1)

--- a/chg/changelog_test.go
+++ b/chg/changelog_test.go
@@ -189,6 +189,31 @@ func TestChangelogReleaseNoOvewriteCompareURL(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestChangelogReleaseParseLinkFromPreviousVersion(t *testing.T) {
+	c := Changelog{
+		Versions: []*Version{
+			{
+				Name: "Unreleased",
+				Link: "https://github.com/org/repo/compare/0.2.0...HEAD",
+				Changes: []*ChangeList{
+					{
+						Type: Added,
+						Items: []*Item{
+							{Description: "New feature"},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	newVersion, err := c.Release(Version{Name: "1.0.0"})
+	assert.NoError(t, err)
+	assert.NotNil(t, newVersion)
+
+	assert.Equal(t, "https://github.com/org/repo/compare/0.2.0...1.0.0", newVersion.Link)
+}
+
 func TestChangelogRenderLinks(t *testing.T) {
 	unreleased := &Version{Name: "Unreleased", Link: "http://example.com/unreleased"}
 	v123 := &Version{Name: "1.2.3", Link: "http://example.com/1.2.3"}

